pkg/controller/apicast/apicast_controller.go

```diff
@@ -189,6 +191,41 @@ func (r *ReconcileAPIcast) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileAPIcast) updateStatus(instance *appsv1alpha1.APIcast, reconciler *APIcastLogicReconciler) (reconcile.Result, error) {
```

Comment:
We've implemented the updateStatus method here in the apicast_controller code.
We could make a public method ReconcileStatus or UpdateStatus inside the APIcastLogicReconciler or other alternative maybe. Maybe it's good enough here for now.

Reply:
Makes sense that status reconciliation is implemented outside `APIcastLogicReconciler`. So maybe, best to go is making `reconciler.APIcastFromCRContents()` helper method to be shared. Can be future improvement